### PR TITLE
feat: Refactor OpenAIFunction to FunctionTool

### LINKED ISSUE #858: [Feature Request] Update naming for OpenAIFunction
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

The class "OpenAIFunction" doesn't actually require OpenAI, it's a legacy name from earlier development.we can renaming it to something more fitting, like "CamelTool."

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tom-doerr**
## Description

I renamed the OpenAIFunction class to FunctionTool and created a new OpenAIFunction class that throws a deprecation warning when initialized.

## Motivation and Context

closes #858 

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Wendong-Fan** [comment]
Thanks @tom-doerr ! We updated the way we calling tools few days ago, could you help update this PR with latest master branch and resolve the conflicts? Thanks in advance!

**Comment by tom-doerr** [comment]
Will finish that today!

**Comment by tom-doerr** [comment]
Update from my side: The merge is more comprehensive than I thought, even after I solved all conflicts the tests don't pass.
```
 ImportError: cannot import name 'OpenAIFunction' from partially initialized module 'camel.toolkits' (most likely due to a circular import
```
I will try again tomorrow to fix this. If that doesn't work I would just redo the changes on top of the newest master version.

**Comment by Wendong-Fan** [comment]
> Update from my side: The merge is more comprehensive than I thought, even after I solved all conflicts the tests don't pass.
> 
> ```
>  ImportError: cannot import name 'OpenAIFunction' from partially initialized module 'camel.toolkits' (most likely due to a circular import
> ```
> 
> I will try again tomorrow to fix this. If that doesn't work I would just redo the changes on top of the newest master version.

Thanks @tom-doerr ! Take your time and if it's hard to be fixed you can also just push the code, I'd love to support further

**Review by Wendong-Fan — APPROVED** [review]
Thanks @tom-doerr 's contribution! I added some commits and now it looks good to me, please check the update and let me know if there's anything need to be further improved, thanks!

File: camel/toolkits/openai_function.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
```
Comment (yiyiyi0817) [inline camel/toolkits/openai_function.py:13]: Do we need to also change the name of this file, such as `function_tool.py`?

**Review by yiyiyi0817 — COMMENTED** [review]
LGTM! I support the renaming. Just leave a question about the detail.

File: camel/toolkits/openai_function.py (hunk 1 above)
Comment (Wendong-Fan) [reply camel/toolkits/openai_function.py:13]: updated

**Pull request merged**